src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusMetric.cs

```diff
@@ -168,6 +170,7 @@ internal static string RemoveAnnotations(string unit)
             return unit;
         }
 
+        sb ??= new StringBuilder();
         sb.Append(unit, lastWriteIndex, unit.Length - lastWriteIndex);
         return sb.ToString();
     }
```

Comment:
if `sb` is null here, can probably avoid the extra allocation by just returning `unit`

Reply:
It seems like the block right above this is really handling the `sb == null` case. I added a `Debug.Assert` here instead.